Gem/Code/Source/Components/Multiplayer/PlayerJumpPadEffectComponent.cpp

```diff
@@ -0,0 +1,100 @@
+    }
+
+    void PlayerJumpPadEffectComponent::OnInit()
+    {
+    }
+
+    void PlayerJumpPadEffectComponent::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void PlayerJumpPadEffectComponent::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+
+    PlayerJumpPadEffectComponentController::PlayerJumpPadEffectComponentController(PlayerJumpPadEffectComponent& parent)
+        : PlayerJumpPadEffectComponentControllerBase(parent)
+    {
+    }
+
+    void PlayerJumpPadEffectComponentController::OnActivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+    }
+
+    void PlayerJumpPadEffectComponentController::OnDeactivate([[maybe_unused]] Multiplayer::EntityIsMigrating entityIsMigrating)
+    {
+        m_jumpPadEffectEvent.RemoveFromQueue();
+    }
+
+    void PlayerJumpPadEffectComponentController::HandleRPC_ApplyJumpPadEffect(
+        [[maybe_unused]] AzNetworking::IConnection* invokingConnection, 
+        const AZ::Vector3& jumpVelocity, const AZ::TimeMs& effectDuration)
+    {
+        GetNetworkTransformComponentController()->ModifyResetCount()++;
+        RPC_StartJump(jumpVelocity, effectDuration);
+    }
+
+    void PlayerJumpPadEffectComponentController::HandleRPC_StartJump([[maybe_unused]] AzNetworking::IConnection* invokingConnection,
+        const AZ::Vector3& jumpVelocity, const AZ::TimeMs& effectDuration)
+    {
+        m_isUnderJumpPadEffect = true;
+        m_jumpPadVector = jumpVelocity;
+        m_jumpPadEffectEvent.Enqueue(effectDuration);
+    }
+
+    void PlayerJumpPadEffectComponentController::JumpPadEffectTick()
+    {
+        // Jump effect is over
+        m_isUnderJumpPadEffect = false;
+    }
+
+    void PlayerJumpPadEffectComponentController::CreateInput([[maybe_unused]] Multiplayer::NetworkInput& input, [[maybe_unused]] float deltaTime)
+    {
+        auto* componentInput = input.FindComponentInput<PlayerJumpPadEffectComponentNetworkInput>();
+        if (componentInput)
+        {
+            componentInput->m_jumpPadEffect = m_isUnderJumpPadEffect;
+            componentInput->m_jumpPadVector = m_jumpPadVector;
+        }
+    }
+
+    void PlayerJumpPadEffectComponentController::ProcessInput([[maybe_unused]] Multiplayer::NetworkInput& input, [[maybe_unused]] float deltaTime)
+    {
+        const auto* componentInput = input.FindComponentInput<PlayerJumpPadEffectComponentNetworkInput>();
+        if (componentInput)
+        {
+            if (componentInput->m_jumpPadEffect)
+            {
+                GetNetworkCharacterComponentController()->TryMoveWithVelocity(componentInput->m_jumpPadVector, deltaTime);
+            }
+            else // apply gravity
+            {                
+                const AZ::Vector3 gravity = AZ::Vector3::CreateAxisZ(-GetGravity());
+                GetNetworkCharacterComponentController()->TryMoveWithVelocity(gravity, deltaTime);
```

Comment:
I'm OK with this for now, but I think we should be sending impulse requests to the player movement component from jump pads/forces etc and then in the player movement controller we do ground checks and apply gravity + the impulses in one place.  I am working on this but I'm in Lua land because it's a lot quicker turn around time and I haven't gotten to the c++ conversion and PR yet.

Reply:
Yeah, as I don't have any gravity right now in the branch, I'm adding here temporarily.